### ISSUE TITLE
Parsing nulls/bools/ints/doubles as strings

### ISSUE THREAD

**Issue by corporateshark** [miloyip/rapidjson#560]
The SAX parser generates separate events for null/bool/ints/int64/double values and parses them accordingly. This is according to the JSON specification. However, in some applications significant performance boost can be achieved by eliminating the conversion stage and treating all the values as strings. A handler can be boiled down to this form:
```
{
    bool String(const Ch*, SizeType, bool) { return static_cast<Override&>(*this).Default(); }
    bool StartObject() { return static_cast<Override&>(*this).Default(); }
    bool Key(const Ch* str, SizeType len, bool copy) { return static_cast<Override&>(*this).String(str, len, copy); }
    bool EndObject(SizeType) { return static_cast<Override&>(*this).Default(); }
    bool StartArray() { return static_cast<Override&>(*this).Default(); }
    bool EndArray(SizeType) { return static_cast<Override&>(*this).Default(); }
}
```

**Comment by miloyip** [comment]
This is possible but what will be the application of this? Roundtrip will not be possible with this as well.

Null/true/false does not boost performance IMHO, validation is still necessary. However, I have thought of providing a parse option, which skips conversion of numbers (just validates) and generates event to a special handler function, like:

~~~cpp
bool Number(const Ch* SizeType len);
~~~

It is possible to make a roundtrip for that. And user can parse the number only when it is needed.

**Comment by corporateshark** [comment]
We use this when JSON is parsed into an intermediate data format inside the app and this format treats everything as strings eliminating variant data types at tree nodes. 

I have a working implementation with a new parsing flag and a relatively small patch to `ParseNumber()`. Can clean it up and open a pool request.

**Comment by h46incon** [comment]
Writer.RawNumber() will write number as string, is this in design?

**Comment by corporateshark** [comment]
Probably not. I guess the Writer should be symmetrical to the Reader and write `int`s in a sense that it should try converting string to int and then write it.

**Comment by h46incon** [comment]
I am trying to fix it but meet problem similar to #1152 .

**Comment by h46incon** [comment]
Similiar issues about quotes of raw number: #704 , #852
